ton_client/src/abi/tests.rs

```diff
@@ -531,3 +535,171 @@ fn test_encode_message_pubkey_internal(
+            +1HbxbPCwDtR28RzxbJ7VRwatswAQm0ZfaLwBYB+O1Hb2FujjvtRNAg10nCAY4W9ATTP9MA7UcBb3EBb3YBb3MBb3Lt\
+            V44Y9AXtRwFvcnBvc3BvdsiAIM9AydBvce1X4t7R7UdvEdcLH8iCEFDL7ReCEIAAAACxzwsfIc8LH8hzzwsB+CjPFnL\
+            PQPglzws/gCHPQCDPNSLPMbwXAHiWcc9AIc8XlXHPQSHN4iDJcfsAWyHA/44e7UdvEsj0AO1HbxPPCz/tR28WzwsA7U\
+            dvEc8Wye1U3nFq2zACASAbGQEJu3MS5FgaAPjtR29hbo477UTQINdJwgGOFvQE0z/TAO1HAW9xAW92AW9zAW9y7VeOG\
+            PQF7UcBb3Jwb3Nwb3bIgCDPQMnQb3HtV+Le+ADR+CO1H+1HIG8RMAHIyx/J0G9R7VftR28SyPQA7UdvE88LP+1HbxbP\
+            CwDtR28RzxbJ7VRwatswAMrdcCHXSSDBII4rIMAAjhwj0HPXIdcLACDAAZbbMF8H2zCW2zBfB9sw4wTZltswXwbbMOM\
+            E2eAi0x80IHS7II4VMCCCEP////+6IJkwIIIQ/////rrf35bbMF8H2zDgIyHxQAFfBw=="
+        )
+    )?;
+
+    test_encode_internal_message_deploy(
+        &client,
+        &image,
+        &abi,
+        &tvc,
+        Some(CallSet {
+            function_name: "constructor".into(),
+            header: None,
+            input: None,
+        }),
+        None,
+    )?;
+
+    test_encode_internal_message_run(
+        &client,
+        &abi,
+        Some(CallSet {
+            function_name: "sayHello".into(),
+            header: None,
+            input: None,
+        }),
+        Some("te6ccgEBAQEANgAAaAIACRorPEhV5veJGis8SFXm94kaKzxIVeb3iRorPEhV5veAAAAAAAAAAAAAAAAAAFDL7Rc="),
+    )
+}
+
+fn test_encode_internal_message_run(
+    client: &TestClient,
+    abi: &Abi,
+    call_set: Option<CallSet>,
+    expected_boc: Option<&str>,
+) -> Result<()> {
+    fn check_str_field(map: &serde_json::Map<String, serde_json::Value>, field_name: &str, expected: &str) {
+        let value = map.get(field_name)
+            .expect(&format!("Expected `{}` field", field_name));
+        let value = value.as_str()
+            .expect(&format!("Expected `{}` to have `String` type", field_name));
+        assert_eq!(value, expected, "Check failed: {}", field_name);
+    }
+
+    let address = String::from("0:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef");
+    let result: ResultOfEncodeInternalMessage = client.request(
+        "abi.encode_internal_message",
+        ParamsOfEncodeInternalMessage {
+            abi: abi.clone(),
+            address: Some(address.clone()),
+            deploy_set: None,
+            call_set,
+            value: "1000000000".to_string(),
+            bounce: Some(true),
+            enable_ihr: None
+        },
+    )?;
+
+    assert_eq!(result.address, address);
+    assert_eq!(result.message_id, get_boc_hash(&base64::decode(&result.message)?)?);
+    if let Some(expected_boc) = expected_boc {
+        assert_eq!(&result.message, expected_boc);
+    }
+
+    let parsed = parse_message(
+        Arc::new(ClientContext::new(Default::default())?),
+        ParamsOfParse {
+            boc: result.message
+        }
+    )?;
+
+    let json_obj = parsed.parsed.as_object()
+        .expect("Expected JSON Object");
+
+    check_str_field(json_obj, "msg_type_name", "internal");
```

Comment:
You could just write `assert_eq!(parsed.parsed["msg_type_name"], "internal");` FYI